read-fonts/src/table_provider.rs

```diff
@@ -12,133 +20,115 @@ pub trait TableProvider<'a> {
         self.data_for_tag(tag).ok_or(ReadError::TableIsMissing(tag))
     }
 
+    fn expect_table<T: TopLevelTable + FontRead<'a>>(&self) -> Result<T, ReadError> {
+        self.expect_data_for_tag(T::TAG).and_then(FontRead::read)
+    }
+
+    fn expect_table_args<T: TopLevelTable + FontReadWithArgs<'a>>(
+        &self,
+        args: &T::Args,
+    ) -> Result<T, ReadError> {
+        self.expect_data_for_tag(T::TAG)
+            .and_then(|data| T::read_with_args(data, args))
+    }
+
     fn head(&self) -> Result<tables::head::Head<'a>, ReadError> {
```

Comment:
At this point do we even need per-table functions? - to me it would feel idiomatic to have something like `fn table` that could return whichever one you want so you can write things like:

```rust
let head: Head = font.table()?
```

Reply:
You'd have to do a little bit of gymnastics to figure out read args but ... I think it might be achievable?

Reply:
I personally think it is a nice convenience to just have a simple method that returns a table, and I definitely think it is good to hide away the `FontReadWithArgs` business.